lib/client/client_store.go

```diff
@@ -0,0 +1,280 @@
+			return nil, trace.BadParameter("no profile provided and no current profile")
+		}
+	} else {
+		// remove ports from proxy host, because profile name is stored by host name
+		profileName, err = utils.Host(profileName)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+
+	profile, err := s.GetProfile(profileName)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	idx := KeyIndex{
+		ProxyHost:   profileName,
+		ClusterName: profile.SiteName,
+		Username:    profile.Username,
+	}
+	key, err := s.GetKey(idx, WithAllCerts...)
+	if err != nil {
+		if trace.IsNotFound(err) {
+			// If we can't find a key to match the profile, return a partial status. This
+			// is used for some superficial functions `tsh logout` and `tsh status`.
+			return &ProfileStatus{
+				Name: profileName,
+				Dir:  profile.Dir,
+				ProxyURL: url.URL{
+					Scheme: "https",
+					Host:   profile.WebProxyAddr,
+				},
+				Username:    profile.Username,
+				Cluster:     profile.SiteName,
+				KubeEnabled: profile.KubeProxyAddr != "",
+				// Set ValidUntil to now to show that the keys are not available.
+				ValidUntil: time.Now(),
+			}, nil
+		}
+		return nil, trace.Wrap(err)
+	}
+
+	_, onDisk := s.KeyStore.(*FSKeyStore)
+
+	return profileStatusFromKey(key, profileOptions{
+		ProfileName:   profileName,
+		ProfileDir:    profile.Dir,
+		WebProxyAddr:  profile.WebProxyAddr,
+		Username:      profile.Username,
+		SiteName:      profile.SiteName,
+		KubeProxyAddr: profile.KubeProxyAddr,
+		IsVirtual:     !onDisk,
+	})
+}
+
+// FullProfileStatus returns the name of the current profile with a
+// a list of all active profile statuses.
+func (s *Store) FullProfileStatus() (*ProfileStatus, []*ProfileStatus, error) {
+	currentProfileName, err := s.CurrentProfile()
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+
+	currentProfile, err := s.ReadProfileStatus(currentProfileName)
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+
+	profileNames, err := s.ListProfiles()
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+
+	var profiles []*ProfileStatus
+	for _, profileName := range profileNames {
+		if profileName == currentProfileName {
+			// already loaded this one
+			continue
+		}
+		status, err := s.ReadProfileStatus(profileName)
+		if err != nil {
+			return nil, nil, trace.Wrap(err)
+		}
+		profiles = append(profiles, status)
+	}
+
+	return currentProfile, profiles, nil
+}
+
+// noClientStore is a ClientStore representing the absence of a ClientStore.
+// All methods return errors. This exists to avoid nil checking everywhere in
+// LocalKeyAgent and prevent nil pointer panics.
+type noClientStore struct{}
+
+func newNoClientStore() *Store {
+	return &Store{
+		log:               logrus.WithField(trace.Component, teleport.ComponentKeyStore),
+		KeyStore:          noClientStore{},
+		TrustedCertsStore: noClientStore{},
+		ProfileStore:      noClientStore{},
+	}
+}
+
+var errNoClientStore = trace.NotFound("there is no client store")
```

Comment:
:nail_care: 

`trace.NotImplemented` seems to fit here, what do you think?

Reply:
`noClientStore` is meant to act as an empty non-nil client store, so we want to return not found errors instead. I'll clarify in the `noClientStore` godoc.